### PR TITLE
feat: bump alloy

### PR DESCRIPTION
**Pull request by klkvr**
Bumps alloy-core, alloy and op-alloy

Related PRs:
https://github.com/alloy-rs/op-alloy/pull/205
https://github.com/alloy-rs/alloy/pull/1540
https://github.com/alloy-rs/core/pull/796

### CODE REVIEW

File: crates/consensus/debug-client/src/client.rs [hunk 1]
```diff
@@ -205,15 +206,10 @@ pub fn block_to_execution_payload_v3(block: Block) -> ExecutionNewPayload {
...
-                        let envelope: TxEnvelope = tx.try_into().unwrap();
-                        let mut buffer: Vec<u8> = vec![];
-                        envelope.encode_2718(&mut buffer);
-                        buffer.into()
-                    })
+                    .map(|tx| tx.inner.encoded_2718().into())
                     .collect(),
             },
-            withdrawals: block.withdrawals.clone().unwrap_or_default(),
+            withdrawals: block.withdrawals.clone().unwrap_or_default().to_vec(),
```
Comment (mattsse) [inline crates/consensus/debug-client/src/client.rs:212]:
```suggestion
            withdrawals: block.withdrawals.clone().unwrap_or_default().into_inner(),
```

File: crates/optimism/rpc/src/eth/transaction.rs [hunk 2]
```diff
@@ -100,22 +110,29 @@ where
...
             .and_then(|receipt| receipt.deposit_receipt_version);
 
+        let TransactionInfo { block_hash, block_number, index: transaction_index, .. } = tx_info;
+
         Transaction {
-            inner,
-            source_hash: signed_tx.source_hash(),
-            mint: signed_tx.mint(),
-            // only include is_system_tx if true: <https://github.com/ethereum-optimism/op-geth/blob/641e996a2dcf1f81bac9416cb6124f86a69f1de7/internal/ethapi/api.go#L1518-L1518>
-            is_system_tx: (signed_tx.is_deposit() && signed_tx.is_system_transaction())
```
Comment (mattsse) [inline crates/optimism/rpc/src/eth/transaction.rs:108]: ah great, this is now solved directly via the field

File: crates/rpc/rpc-types-compat/src/proof.rs [hunk 3]
```diff
@@ -6,7 +6,11 @@ use reth_trie_common::{AccountProof, StorageProof};
 
 /// Creates a new rpc storage proof from a primitive storage proof type.
 pub fn from_primitive_storage_proof(proof: StorageProof) -> EIP1186StorageProof {
-    EIP1186StorageProof { key: JsonStorageKey(proof.key), value: proof.value, proof: proof.proof }
+    EIP1186StorageProof {
+        key: JsonStorageKey::Hash(proof.key),
+        value: proof.value,
+        proof: proof.proof,
+    }
 }
```
Comment (mattsse) [inline crates/rpc/rpc-types-compat/src/proof.rs:14]: Idea for followup, should we move those alloy types into alloy-eips?

**Review by mattsse — APPROVED** [review]
wow, this simplifies things a lot

File: crates/rpc/rpc-types-compat/src/proof.rs (hunk 3 above)
Comment (klkvr) [reply crates/rpc/rpc-types-compat/src/proof.rs:14]: eip1186 mod might make sense, though this EIP is basically just RPC spec so probably would still be closer to RPC types

**Pull request merged**